File: engine/asset/asset.cpp
#include "asset/asset.h"

std::string serialize_asset_type(const AssetType type) {
	switch (type) {
		case AssetType::TEXTURE:
			return "TEXTURE";
		case AssetType::FONT:
			return "FONT";
		case AssetType::SCENE:
			return "SCENE";
		default:
			return "";
	}
}

File: script_core/asset/asset.cs
namespace EveEngine
{
	public enum AssetType
	{
		None,
		Texture,
		Font,
		Scene
	}

	public abstract class Asset
	{
		public readonly ulong handle;
		public readonly string path;
		public readonly AssetType type;

		public Asset(ulong handle, string path, AssetType type)
		{
			this.handle = handle;
			this.path = path;
			this.type = type;
		}

		public Asset(string path, AssetType type)
		{
			this.path = path;
			this.type = type;

			this.handle = AssetRegistry.Load(path, type);
		}
	}
}
